### PR TITLE
Terraform init is required before apply

### PR DESCRIPTION
**Pull request by wandergeek**
## Description
Just a doco change, noticed when running through examples.

Are these examples supposed to spin up enterprise search as well? Found it confusing when I hit the UI and saw lots of "Getting started with Enterprise Search" text. 

## Types of Changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (improves code quality but has no user-facing effect)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation

### CODE REVIEW

File: examples/deployment/README.md [hunk 1]
```diff
@@ -5,4 +5,4 @@ The created resources are a single-node Elasticsearch cluster with a Kibana, APM
 
 ## Running the example
 
-run `terraform apply` to see it work.
+run `terraform init && terraform apply` to see it work.
```
Comment (karencfv) [inline examples/deployment/README.md:8]:
```suggestion
Build the provider using `make install` from the main folder. From within the example's directory run `terrafrom init` to initialize Terrafrom, and `terraform apply` to see it work.
```

File: examples/deployment_ccs/README.md [hunk 2]
```diff
@@ -5,4 +5,4 @@ The created resources are: two single node Elasticsearch clusters acting as sour
 
 ## Running the example
 
-run `terraform apply` to see it work.
+run `terraform init && terraform apply` to see it work.
```
Comment (karencfv) [inline examples/deployment_ccs/README.md:8]:
```suggestion
Build the provider using `make install` from the main folder. From within the example's directory run `terrafrom init` to initialize Terrafrom, and `terraform apply` to see it work.
```

File: examples/README.md [hunk 3]
```diff
@@ -2,4 +2,4 @@
 
 This directory contains a set of examples of using various Elastic Cloud resources with Terraform. The examples each have their own README containing more details on what the example does.
 
-To run any example, clone the repository and run `terraform apply` within the example's own directory.
+To run any example, clone the repository and run `terraform init && terraform apply` within the example's own directory.
```
Comment (karencfv) [inline examples/README.md:5]:
```suggestion
To run any example clone the repository. Build the provider using `make install` from the main folder. From within the example's directory run `terrafrom init` to initialize Terrafrom, and `terraform apply` to see it work.

```

**Review by karencfv — COMMENTED** [review]
Thanks for the PR! 🎉  Just added some suggestions so our docs are a bit clearer for people with different levels of technical knowledge ;)

File: examples/README.md [hunk 4]
```diff
@@ -2,4 +2,4 @@
 
 This directory contains a set of examples of using various Elastic Cloud resources with Terraform. The examples each have their own README containing more details on what the example does.
 
-To run any example, clone the repository and run `terraform apply` within the example's own directory.
+To run any example clone the repository. Build the provider using `make install` from the main folder. From within the example's directory run `terrafrom init` to initialize Terrafrom, and `terraform apply` to see it work.
```
Comment (alaudazzi) [inline examples/README.md:5]:
```suggestion
To run any example, clone the repository. Build the provider using `make install` from the main folder. From within the example's directory, run `terraform init` to initialize Terraform, and `terraform apply` to see if it works.
```

File: examples/deployment/README.md [hunk 5]
```diff
@@ -5,4 +5,4 @@ The created resources are a single-node Elasticsearch cluster with a Kibana, APM
 
 ## Running the example
 
-run `terraform apply` to see it work.
+Build the provider using `make install` from the main folder. From within the example's directory run `terrafrom init` to initialize Terrafrom, and `terraform apply` to see it work.
```
Comment (alaudazzi) [inline examples/deployment/README.md:8]:
```suggestion
Build the provider using `make install` from the main folder. From within the example's directory, run `terraform init` to initialize Terraform, and `terraform apply` to see if it works.
```

File: examples/deployment_ccs/README.md [hunk 6]
```diff
@@ -5,4 +5,4 @@ The created resources are: two single node Elasticsearch clusters acting as sour
 
 ## Running the example
 
-run `terraform apply` to see it work.
+Build the provider using `make install` from the main folder. From within the example's directory run `terrafrom init` to initialize Terrafrom, and `terraform apply` to see it work.
```
Comment (alaudazzi) [inline examples/deployment_ccs/README.md:8]:
```suggestion
Build the provider using `make install` from the main folder. From within the example's directory, run `terraform init` to initialize Terraform, and `terraform apply` to see if it works.
```

**Review by alaudazzi — APPROVED** [review]
Fixed a few typos. Otherwise LGTM.

File: examples/README.md (hunk 4 above)
Comment (karencfv) [reply examples/README.md:5]: @alaudazzi I'm not sure how I feel about the phrase `to see if it works` 😄 . To me it sounds like we are giving the impression that our example may or may not work 🤔. The initial intent of the original sentence was to say something like "Run this to see it in action" or to witness it working. 

Do you think maybe there is another way to phrase your suggestion for this sentence?

Comment (karencfv) [reply examples/README.md:5]: I just saw you changed it for `to see how it works` here https://github.com/elastic/terraform-provider-ec/pull/186/files#diff-150772696d1ea70df8dc2f6fe2c564dd3b72d5b93efc46f9baf9009333d60f00R8

That works for me!

**Comment by karencfv** [comment]
@wandergeek heya! 👋 looks like there are some conflicts with the main branch now, could you fix those and address @alaudazzi 's comments so we can get this merged? :)

File: examples/deployment/README.md (hunk 5 above)
Comment (karencfv) [reply examples/deployment/README.md:8]: same `to see how it works`

File: examples/deployment_ccs/README.md (hunk 6 above)
Comment (karencfv) [reply examples/deployment_ccs/README.md:8]: same `to see how it works`

**Review by karencfv — APPROVED** [review]

**Pull request merged**